server/command_debug_packet.go

```diff
@@ -0,0 +1,98 @@
+package main
+
+import (
+	"fmt"
+	"time"
+
+	cloud "github.com/mattermost/mattermost-cloud/model"
+	"github.com/mattermost/mattermost-server/v6/model"
+	"github.com/pkg/errors"
+)
+
+func (p *Plugin) runGetDebugPacketCommand(args []string, extra *model.CommandArgs) (*model.CommandResponse, bool, error) {
+	if len(args) == 0 {
+		return nil, true, errors.New("must provide an installation name")
+	}
+
+	name := standardizeName(args[0])
+	if name == "" {
+		return nil, true, errors.New("must provide an installation name")
+	}
+
+	installsForUser, err := p.getInstallationsForUser(extra.UserId)
+	if err != nil {
+		return nil, false, err
+	}
+
+	var installToExec *Installation
+	for _, install := range installsForUser {
+		if install.OwnerID == extra.UserId && install.Name == name {
+			installToExec = install
+			break
+		}
+	}
+
+	if installToExec == nil {
+		return nil, true, fmt.Errorf("no installation with the name %s found", name)
+	}
+
+	p.API.SendEphemeralPost(extra.UserId, &model.Post{
+		UserId:    p.BotUserID,
+		ChannelId: extra.ChannelId,
+		Message:   fmt.Sprintf("Gathering debug data for `%s` now. Please wait as this may take a while.", installToExec.Name),
+	})
+
+	err = p.execGetDebugPacket(installToExec.ID, extra.UserId, name)
+	if err != nil {
+		return nil, false, err
+	}
+
+	resp := "Debug packet generated. Check your direct messages from the cloud bot."
+
+	return getCommandResponse(model.CommandResponseTypeEphemeral, resp, extra), false, nil
```

Comment:
Should we wrap the exec in a go routine, so that a pprof that takes too long to generate won't run into timeout issues from Nginx?

Reply:
Hmm, this is an interesting thought. So I was about to make this change, but I think it may have a trade-off on user confusion depending on what goes wrong. In particular, if the goroutine runs into an issue it needs to perform the lookups to send the user a DM letting them know. Although it's unlikely, those lookups could also fail for some reason and the user would be left in the dark.

In my testing on grabbing the packet it took only a second or two to return. I wonder if we start here and make changes if timeouts actually become an issue?

Reply:
I've not generated one of these before so wasn't sure how long they would take on average. If they're generally pretty quick then this sounds good to me for now!